app/src/main/java/org/connectbot/ui/screens/console/ConsoleScreen.kt

```diff
@@ -341,22 +344,28 @@ fun ConsoleScreen(
                                 bottom = if (keyboardAlwaysVisible) TERMINAL_KEYBOARD_HEIGHT_DP.dp else 0.dp
                             )
                     ) {
-                        val typeface = Typeface.MONOSPACE
-                        // TODO: Make this configurable via downloading fonts!
-//                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-//                            Typeface.CustomFallbackBuilder(
-//                                FontFamily.Builder(
-//                                    Font.Builder(context.assets, "fonts/PowerlineExtraSymbols.ttf").build()
-//                                ).build()
-//                            ).setSystemFallback("monospace").build()
-//                        } else {
-//                            Typeface.MONOSPACE
-//                        }
+                        // Get font: per-host override takes precedence over global default
+                        val globalFontFamily = prefs.getString("fontFamily", "SYSTEM_DEFAULT") ?: "SYSTEM_DEFAULT"
```

Comment:
This should be reading from the profile. I'd also like to keep the fallback building done in the removed comment. The fallback font should be `Typeface.MONOSPACE` before going to the system default. Typically the system fallback font is not a monospace font.

What I'd like to have happen is that a user can compose two fonts in a profile. This is because things like [Powerline Extra Symbols](https://github.com/ryanoasis/powerline-extra-symbols) can usually be downloaded separately from an entire font. Then the `Typeface.CustomFallbackBuilder` might look like:

    Powerline Extra Symbols → Fira Code → monospace

The UI for configuring that might get extremely complicated, so I'm happy with either only allowing two fonts or a checkbox that says something like "Use Powerline Extra Symbols"

Reply:
Partially fixed. ConsoleScreen now reads font only from the profile via bridge.fontFamily. Removed the host-level font override and global preference fallback logic. 
Added scaffolding for buildTypefaceWithFallback() in TerminalFontCompose. The Full CustomFallbackBuilder implementation is limited because the Font.Builder API doesn't accept an existing Typeface directly - it requires raw font data (from assets or resources). The current implementation returns the loaded typeface, and Android's text system provides a natural fallback for missing glyphs. If we want full control over the fallback chain, we'd need to load fonts from files/assets rather than via the Google Fonts API. 

For a locally downloaded font, we can do the composable font, I guess, but it's still not implemented.

Reply:
Although I believe so, adding the functionality might not break other functionality, so now working on this. If there are significant obstacles, may be I will give up, but still I do not observe such things.

Reply:
Well, I found that CustomFallbackBuilder does not work for such character-level replacement. So, this functionality is not implementable at the connectbot level. We need a patch for termlib, but I'm not going to include that in this PR. In short, I give up this time......